xayn-ai/src/ltr/list_net/mod.rs

```diff
@@ -704,15 +753,15 @@ pub trait DataSource {
     /// Returns the next batch of training samples.
     ///
     /// Returns a empty vector once all training samples have been returned.
```

Comment:
```suggestion
    /// Returns an empty vector once all training samples have been returned.
```
while we're here... the docstring above the `reset` function should read
```rust
    /// This is called at the *beginning* of every epoch.
    fn reset(&mut self) -> Result<(), Self::Error>;
```